src/aequitas/bias.py

```diff
@@ -11,52 +16,96 @@
-        else:
-            self.input_group_metrics = input_group_metrics
+        self.key_columns = list(key_columns)
+        self.input_group_metrics = list(input_group_metrics)
+
         if not fill_divbyzero:
             self.fill_divbyzero = 10.00000
         else:
             self.fill_divbyzero = fill_divbyzero
+        self.non_attr_cols = non_attr_cols
+        self.significance_cols = significance_cols
+        # if sample_df:
+        #     self.samples = self.get_measure_sample(sample_df, attribute, measure)
 
-    def get_disparity_min_metric(self, df, key_columns=None, input_group_metrics=None,
-                                 fill_divbyzero=None):
+    def get_disparity_min_metric(self, df, original_df, key_columns=None,
+                                 input_group_metrics=None, fill_divbyzero=None,
+                                 check_significance=None,  alpha = 5e-2,
+                                 mask_significance = True, label_score_ref='fpr'):
         """
-            Calculates several ratios using the group metrics value and dividing by the minimum
-            group metric value among all groups defined by each attribute
+        Calculates several ratios using the group metrics value and dividing by
+        the minimum group metric value among all groups defined by each attribute
+
         :param df: the resulting dataframe from the group get_crosstabs
-        :param input_group_metrics: the columns list corresponding to the group metrics for each
-        we want to calculate the Disparity values
-        :return: a dataframe
+        :param original_df: a dataframe containing a required 'score 'column
+            and possible 'label_value' column.
+        :param key_columns: optional, the key columns to use on joins.
+            Defualts are 'model_id', 'score_threshold', 'attribute_name'.
+        :param input_group_metrics: optional, the columns list corresponding to
+            the group metrics for which we want to calculate disparity values
+        :param check_significance: Measures for which to determine statistical
+            significance beyond label_value and score. Defaults are 'fpr' and 'fnr'.
+        :param alpha: Level at which to determine statistical significance.
+            Default is 5e-2 (0.05).
+        :param mask_significance: Whether to display a T/F mask over calculated
+            p-values from statistical significance determination. Default is True.
+
+        :return: a dataframe with same number of rows as the input (crosstab)
+            but with additional disparity metrics columns, ref_group_values, and
+            statisitcal significance of specific metrics.
         """
 
         print('get_disparity_min_metric()')
         if not key_columns:
-            key_columns = self.key_columns
+            key_columns = [col for col in self.key_columns if col in df.columns]
```

Comment:
Perhaps more efficient or more to-the-point?

```suggestion
            key_columns = df.columns.intersection(self.key_columns).tolist()
```

Reply:
Would this return a list of consistent order? This was actually my first choice implementation but I had concerns about column order with the set operation

Reply:
Good question :smile: 

I don't really know – it appears to do so, but in this case, ordering by `df.columns`. If that's what you want, that should do.